cmake/kokkos_enable_options.cmake

```diff
@@ -85,3 +57,40 @@ ELSE()
 ENDIF()
 KOKKOS_ENABLE_OPTION(CUDA_LAMBDA ${LAMBDA_DEFAULT} "Whether to activate experimental laambda features")
 
+KOKKOS_OPTION(OPTIONS "" STRING "A list of options to enable")
+IF (KOKKOS_OPTIONS MATCHES ",")
+  MESSAGE(WARNING "-- Detected a comma in: Kokkos_OPTIONS=`${KOKKOS_OPTIONS}`")
+  MESSAGE("-- Although we prefer Kokkos_OPTIONS to be semicolon-delimited, we do allow")
+  MESSAGE("-- comma-delimited values for compatibility with scripts (see github.com/trilinos/Trilinos/issues/2330)")
+  STRING(REPLACE "," ";" KOKKOS_OPTIONS "${KOKKOS_OPTIONS}")
+  MESSAGE("-- Commas were changed to semicolons, now Kokkos_OPTIONS=`${KOKKOS_OPTIONS}`")
+ENDIF()
+
+FOREACH(OPT ${KOKKOS_OPTIONS})
+STRING(TOUPPER ${OPT} UC_NAME)
+SET(ENABLE_FLAG ON)
+# strip enable/disable prefix and validate option name
+SET(OPTION_NAME ${UC_NAME})
+STRING(REPLACE ENABLE_ "" OPTION_NAME ${OPTION_NAME})
+STRING(REPLACE DISABLE_ "" OPTION_NAME ${OPTION_NAME})
+IF(NOT OPTION_NAME IN_LIST KOKKOS_VALID_OPTIONS)
+    MESSAGE(SEND_ERROR "'${OPTION_NAME}' entry in Kokkos_OPTIONS is not a valid option")
+ENDIF()
```

Comment:
Moved the block `FOREACH()` options after the declaration of the valid options and their default value and added the six lines above.